cleanDockerImages/cleanDockerImages.groovy

```diff
@@ -0,0 +1,299 @@
+
+import java.util.concurrent.TimeUnit
+
+import static org.artifactory.repo.RepoPathFactory.create
+
+//This is executed via REST API
+// usage: curl -X POST http://localhost:8088/artifactory/api/plugins/execute/cleanDockerImages
+
+// Add your Artifactory local docker repos in the array below
+String[] dockerRepos = ["docker-local-repo"]
+maxDaysProperty = "docker.label.com.jfrog.artifactory.retention.maxDays"
+maxCountProperty = "docker.label.com.jfrog.artifactory.retention.maxCount"
+
+executions{
+    cleanDockerImages() {
+
+        //Reset Global Variables
+        foundManifest=false
+        listEndReached=false
+        imagesCount = new TreeMap<String, Integer>()
+        imagesPathMap = new HashMap<String, List<RepoPath>>()
+
+        dockerRepos.each {
+            log.debug "Call Delete Docker Images: $it"
+            buildParentRepoPaths create(it)
+        }
+        message = '{"Status":"okay"}'
+        status = 200
+    }
+
+}
+
+
+def private void buildParentRepoPaths(RepoPath artifactoryRepoPath){
+
+    def List<RepoPath> grandParentReposList = new ArrayList()
+    simpleTraverse(repositories.getItemInfo(artifactoryRepoPath), grandParentReposList)
+    removeImagesMaxCount(imagesPathMap)
+
+}
+
+
+def private void removeImagesMaxCount(Map repoMap){
+
+    Iterator repoIterator = repoMap.keySet().iterator()
+    while(repoIterator.hasNext()){
+        def String key = repoIterator.next()
+        def ArrayList repoList = repoMap.get(key)
+        def int maxImagesCount = imagesCount.get(key)
+
+        if(maxImagesCount > 0){
+
+
+            //log.debug " ARRAY LIST  Size is " + repoList.size() + " AND MAX COUNT IS " + maxImagesCount
+
+            if(repoList.size() > maxImagesCount){
+
+                // If number of current docker images are more than maxcount, delete them
+                for(int i=0;i<repoList.size()-maxImagesCount;i++){
+                    RepoPath r = repoList.get(i)
+                    log.debug "HASH MAP KEY = " + key + " AND VALUE = " + r.getId()
+
+                    //Logic to delete docker images based on Max Count policy
+                    //Uncomment the below line for the images to actually delete
+                    //deleteDockerImage(RepoPath)
+
+                }
+            }
+
+        }
+
+    }
+
+}
+
+// For debug only
+def private void printImagesMaxCount(Map repoMap){
+
+    Iterator repoIterator = repoMap.keySet().iterator()
+
+    while(repoIterator.hasNext()){
+        def String key = repoIterator.next()
+        def Integer value = repoMap.get(key)
+        //log.debug " HASH MAP KEY = " + key + " and VALUE = " + value
+
+    }
+
+}
+
+
+/*
+ * Traverse through the docker repo(directories and sub-directories) and
+ * 1) delete the images immediately if the maxDays policy applies
+ * 2) Aggregate the images that qualify for maxCount policy (to get deleted in the execution closure)
+ *
+ */
+def private void simpleTraverse(ItemInfo parentInfo, List revistFoldersList){
```

Comment:
Why pass `revistFoldersList` into this function? You never seem to use its value.